ema_workbench/em_framework/samplers.py

```diff
@@ -350,11 +394,75 @@ def __iter__(self) -> Iterable[Point]:
     def __str__(self):  # noqa: D105
         return f"ema_workbench.DesignIterator, {self.n} designs on {len(self.params)} parameters"
 
+    def combine(
+        self,
+        other: "DesignIterator",
+        combine: Literal["full_factorial", "sample", "cycle"],
+        kind: type[Point]|None = None,
+        rng: SeedLike | RNGLike | None = None,
+    ) -> "DesignIterator":
+        """Combine 2 design iterators into a new design iterator..
+
+        Parameters
+        ----------
+        other : the iterator to combine with this one
+        combine : how to combine the designs.
+        kind : type[Point]|None
+               the Point sublcass to use for the return iterator
+               in case of None, it will use the same kind if both iterators have the same otherwise
+               it will just fall back on using Point.
+        rng : RNG or None, only relevant in case combine is "sample"
+
+        Returns
+        -------
+        a new DesignIterator instance
 
-def design_generator(samples:np.ndarray, params:list[Parameter], kind:type[Point]):
+        """
+        combined_samples = None
+        samples_1 = self.samples
+        samples_2 = other.samples
+
+        match combine:
+            case "full_factorial":
+                samples_1_repeated = np.repeat(
+                    samples_1, repeats=samples_2.shape[0], axis=0
+                )
+                samples_2_tiled = np.tile(samples_2, (samples_1.shape[0], 1))
+                combined_samples = np.hstack((samples_1_repeated, samples_2_tiled))
```

Comment:
```Python
combined_list = [np.concatenate([s1, s2]) for s1, s2 in itertools.product(samples_1, samples_2)]
combined_samples = np.array(combined_list)
```

Reply:
They are well optimized for normal Python objects, but numpy is a special beast. In particular, going from numpy to lists and back again is comparatively slow.